File: 14.longest-common-prefix/nodejs/tests/longest_common_prefix.test.js
const { Solution } = require('../src/longest_common_prefix')

describe('../src/longest_common_prefix.js', () => {
	it("Test: have longest common prefix", () => {
		const input = [ "flower", "flow", "flight" ]
		const expected = "fl"
	
		processTestCase(input, expected)
	})

	it("Test: no longest common prefix", () => {
		const input = [ "dog", "racecar", "car" ]
		const expected = ""
	
		processTestCase(input, expected)
	})
})

const solution = new Solution()

const processTestCase = function (input, expected) {
	const actual = solution.longestCommonPrefix(input)
	expect(actual).toBe(expected)
}

File: 14.longest-common-prefix/golang/tests/longest_common_prefix_test.go
package tests

import (
	"github.com/stretchr/testify/assert"
	"leetcode/src" // follow the project structure
	"testing"
)

func processTestCase(t *testing.T, input []string, expected string) {
	actual := solution.LongestCommonPrefix(input)
	assert.Equal(t, expected, actual)
}

func Test_have_longest_common_prefix(t *testing.T) {
	input := []string{"flower", "flow", "flight"}
	expected := "fl"

	processTestCase(t, input, expected)
}

func Test_no_longest_common_prefix(t *testing.T) {
	input := []string{"dog", "racecar", "car"}
	expected := ""

	processTestCase(t, input, expected)
}
